src/interactions/forbidClickInteraction.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  AbstractInteraction,
+  EventType,
+  VcsEvent,
+  ModificationKeyType,
+} from '@vcmap/core'
+import type { InteractionEvent } from '@vcmap/core'
+import { useViewsStore } from '@/stores/views'
+import { viewList } from '@/model/views.model'
+import { useHomeStore } from '@/stores/home'
+
+class ForbidenClickInteraction extends AbstractInteraction {
```

Comment:
Instead of creating a new interaction only for the home we can keep the main interaction: selectRoofInteraction . And instead of putting the _isClickOnHomePageValid() function in selectRoofInteraction we can put it in the interactionUtils file. And in the selectRoofInteraction interaction add a function that we can for example call: checkViewConditions. And this function brings together the sets of special conditions that must be checked when clicking on a building in the different views.
What do you think ?

Reply:
There was a misunderstanding here. On the home page, when the user clicks on the map if the checkbox is not checked, the error message is displayed. And that's all . The click on the building does not change page.